src/bootstrap/src/core/builder/tests.rs

```diff
@@ -1644,10 +1644,31 @@ mod snapshot {
         ");
     }
 
+    #[test]
+    fn test_lld_opt_in() {
+        let target: &'static str = Box::leak(Box::new(host_target()));
+        let slice: &'static [&'static str] = Box::leak(Box::new([target]));
+
+        with_lld_opt_in_targets(slice, || {
+            let ctx = TestCtx::new();
+
+            insta::assert_snapshot!(
+                ctx.config("doc")
```

Comment:
Let's do `build compiler` here, and get rid of `override_target_no_std`. We don't need to check `doc` here, we just want to see that `LldWrapper` was built.

You can also call `create_config()` on the config builder and assert that LLD was indeed enabled.